tests/deploy/e2e/test_autostarted.py

```diff
@@ -1332,14 +1332,26 @@ async def wait_for_down_status() -> bool:
     }
     assert actual_data[0] == expected_data
 
-    await client.all_agents_action(tid=environment, action=AgentAction.pause.value)
-
+    # Let's restart everything and check that the resource is considered as available
     snippetcompiler.setup_for_snippet(model, ministd=True, index_url="https://pypi.org/simple")
     version, res, status = await snippetcompiler.do_export_and_deploy(include_status=True)
     result = await client.release_version(environment, version, push=False)
     assert result.code == 200
 
-    # Let's restart everything and check that the resource is considered as available
+    async def wait_for_up_status() -> bool:
+        """
+        Wait for the up status of Agent1
+        """
+        result = await client.get_agents(environment)
+        assert result.code == 200
+        actual_data = result.result["data"]
+        if len(actual_data) != 1:
+            return False
+        return actual_data[0]["status"] == "up"
+
+    # Wait for the scheduler to be up and set the status to available
+    await retry_limited(wait_for_up_status, timeout=5)
+
```

Comment:
The previous method did not work because `release_version` did not wait for paused agents to be up

Reply:
*this method is still not working, there is a race condition between the scheduler restarting, it setting the state to available and it scheduling the resource to be deployed